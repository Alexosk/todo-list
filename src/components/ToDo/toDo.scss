@import 'src/styles/functions.scss';

.todo-list {
  margin: space(8) 0;
  max-height: space(100);
  overflow-y: auto;

  .todo-item {
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;
    margin-bottom: space(4);
    padding: space(1) space(4);
    min-height: space(11);
    background: #d66d75;
    border-radius: space(1);
    cursor: default;
    overflow: hidden;

    @media (hover: hover) {
      transition: background-color 200ms, color 200ms, border 200ms;

      &:hover {
        opacity: 0.7;

        .action-button {
          transition: right 250ms;
          right: space(14);
        }
      }
    }

    p {
      display: -webkit-box;
      margin: 0;
      height: 100%;
      max-width: 90%;
      cursor: pointer;
      font-size: 18px;
      overflow: hidden;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;

      &.is-strike-through {
        text-decoration: line-through;
      }
    }

    .action-button {
      display: flex;
      align-items: center;
      position: absolute;
      right: space(-14);
      height: 100%;

      &.is-mobile {
        right: space(14);
      }
    }
  }
}
