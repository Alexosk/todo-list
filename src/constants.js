export const dummyArray = [
  {
    _id: 1,
    task: 'Clean apartment',
    done: false,
  },
  {
    _id: 2,
    task: 'Do laundry',
    done: true,
  },
  {
    _id: 3,
    task: 'Vacuum floor',
    done: false,
  },
  {
    _id: 4,
    task: 'Feed cat',
    done: true,
  },
  {
    _id: 5,
    task: 'Change light bulbs',
    done: false,
  },
  {
    _id: 6,
    task: 'Go to Store',
    done: true,
  },
];
