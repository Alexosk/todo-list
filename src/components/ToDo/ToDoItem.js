import React from 'react';
import { Button } from '../Button';
import Trash from '../../img/trash.png';
import { detectBrowser, capitalizeFirstLetter } from '../../utils/helpers';

import './toDo.scss';

const ToDoItem = ({ todo, handleRemoveItem, handleUpdateItem }) => {
  const todoId = todo._id;
  const isMobile = detectBrowser() === 'is-mobile';

  return (
    <div className="todo-item">
      <p onClick={() => handleUpdateItem(todoId)} className={todo.done ? 'is-strike-through' : ''}>
        {capitalizeFirstLetter(todo.task)}
      </p>
      <div className={`action-button ${isMobile ? 'is-mobile' : ''}`}>
        <Button onClick={() => handleRemoveItem(todoId)} img={Trash} label="remove" className="is-remove-btn" />
      </div>
    </div>
  );
};

export { ToDoItem };
