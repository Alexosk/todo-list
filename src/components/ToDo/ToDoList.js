import React from 'react';
import { deleteTodo } from '../../Endpoints/Endpoints';
import ToDoItem from './ToDoItem';
import { updateTodo } from '../../Endpoints/Endpoints';
import { findTodoItem } from '../../utils/helpers';

const ToDoList = ({ todos, onTodoDeleteStateUpdate, onTodoUpdateStateUpdate }) => {
  const handleRemoveItem = (id) => {
    if (!id) return;
    const todoItem = findTodoItem(id, todos);
    deleteTodo(todoItem).then(() => {
      onTodoDeleteStateUpdate(todoItem);
    });
  };

  const handleUpdateItem = (id) => {
    if (!id) return;

    const todoItem = findTodoItem(id, todos);
    todoItem.done = !todoItem.done;

    updateTodo(todoItem).then(() => {
      onTodoUpdateStateUpdate(todos);
    });
  };

  return (
    <div className="todo-list">
      {todos &&
        todos.map((todo) => {
          return (
            <ToDoItem
              key={todo._id}
              todo={todo}
              handleRemoveItem={handleRemoveItem}
              handleUpdateItem={handleUpdateItem}
            />
          );
        })}
    </div>
  );
};

export default ToDoList;
