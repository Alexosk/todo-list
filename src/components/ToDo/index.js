import ToDoList from './ToDoList';
import ToDoItem from './ToDoItem';

export { ToDoList, ToDoItem };
