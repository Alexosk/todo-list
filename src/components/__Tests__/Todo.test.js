import React from 'react';
import { ToDoItem } from '../ToDo/ToDoItem';
import { ToDoList } from '../ToDo/ToDoList';
import { dummyArray } from '../../constants';

import { render } from '@testing-library/react';
import { uuid } from 'uuidv4';

const testTodo = {
  id: 1,
  task: 'Eat dinner',
  done: false,
};

describe('TodoItem', () => {
  it('Should render', () => {
    const renderContent = <ToDoItem todo={testTodo} handleRemoveItem={void 0} handleUpdateItem={void 0} />;
    const view = render(renderContent);
    expect(view).toMatchSnapshot();
  });
});

describe('TodoList', () => {
  it('Should render', () => {
    const renderContent = (
      <ToDoList key={uuid()} todos={dummyArray} onTodoDeleteStateUpdate={void 0} onTodoUpdateStateUpdate={void 0} />
    );
    const view = render(renderContent);
    expect(view).toMatchSnapshot();
  });
});
