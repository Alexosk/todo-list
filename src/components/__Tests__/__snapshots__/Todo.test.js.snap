// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo-item"
      >
        <p
          class=""
        >
          Eat dinner
        </p>
        <div
          class="action-button "
        >
          <button
            class="button is-remove-btn"
          >
            <img
              alt="label"
              src="trash.png"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo-item"
    >
      <p
        class=""
      >
        Eat dinner
      </p>
      <div
        class="action-button "
      >
        <button
          class="button is-remove-btn"
        >
          <img
            alt="label"
            src="trash.png"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TodoList Should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo-list"
      >
        <div
          class="todo-item"
        >
          <p
            class=""
          >
            Clean apartment
          </p>
          <div
            class="action-button "
          >
            <button
              class="button is-remove-btn"
            >
              <img
                alt="label"
                src="trash.png"
              />
            </button>
          </div>
        </div>
        <div
          class="todo-item"
        >
          <p
            class="is-strike-through"
          >
            Do laundry
          </p>
          <div
            class="action-button "
          >
            <button
              class="button is-remove-btn"
            >
              <img
                alt="label"
                src="trash.png"
              />
            </button>
          </div>
        </div>
        <div
          class="todo-item"
        >
          <p
            class=""
          >
            Vacuum floor
          </p>
          <div
            class="action-button "
          >
            <button
              class="button is-remove-btn"
            >
              <img
                alt="label"
                src="trash.png"
              />
            </button>
          </div>
        </div>
        <div
          class="todo-item"
        >
          <p
            class="is-strike-through"
          >
            Feed cat
          </p>
          <div
            class="action-button "
          >
            <button
              class="button is-remove-btn"
            >
              <img
                alt="label"
                src="trash.png"
              />
            </button>
          </div>
        </div>
        <div
          class="todo-item"
        >
          <p
            class=""
          >
            Change light bulbs
          </p>
          <div
            class="action-button "
          >
            <button
              class="button is-remove-btn"
            >
              <img
                alt="label"
                src="trash.png"
              />
            </button>
          </div>
        </div>
        <div
          class="todo-item"
        >
          <p
            class="is-strike-through"
          >
            Go to Store
          </p>
          <div
            class="action-button "
          >
            <button
              class="button is-remove-btn"
            >
              <img
                alt="label"
                src="trash.png"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo-list"
    >
      <div
        class="todo-item"
      >
        <p
          class=""
        >
          Clean apartment
        </p>
        <div
          class="action-button "
        >
          <button
            class="button is-remove-btn"
          >
            <img
              alt="label"
              src="trash.png"
            />
          </button>
        </div>
      </div>
      <div
        class="todo-item"
      >
        <p
          class="is-strike-through"
        >
          Do laundry
        </p>
        <div
          class="action-button "
        >
          <button
            class="button is-remove-btn"
          >
            <img
              alt="label"
              src="trash.png"
            />
          </button>
        </div>
      </div>
      <div
        class="todo-item"
      >
        <p
          class=""
        >
          Vacuum floor
        </p>
        <div
          class="action-button "
        >
          <button
            class="button is-remove-btn"
          >
            <img
              alt="label"
              src="trash.png"
            />
          </button>
        </div>
      </div>
      <div
        class="todo-item"
      >
        <p
          class="is-strike-through"
        >
          Feed cat
        </p>
        <div
          class="action-button "
        >
          <button
            class="button is-remove-btn"
          >
            <img
              alt="label"
              src="trash.png"
            />
          </button>
        </div>
      </div>
      <div
        class="todo-item"
      >
        <p
          class=""
        >
          Change light bulbs
        </p>
        <div
          class="action-button "
        >
          <button
            class="button is-remove-btn"
          >
            <img
              alt="label"
              src="trash.png"
            />
          </button>
        </div>
      </div>
      <div
        class="todo-item"
      >
        <p
          class="is-strike-through"
        >
          Go to Store
        </p>
        <div
          class="action-button "
        >
          <button
            class="button is-remove-btn"
          >
            <img
              alt="label"
              src="trash.png"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
